src/libcore/slice/mod.rs

```diff
@@ -1575,6 +1575,134 @@ impl<T> [T] {
         sort::quicksort(self, |a, b| f(a).lt(&f(b)));
     }
 
+    /// Rearranges the elements of the slice such that the following properties hold:
+    ///
+    /// - The value at position `index` will be the same as if the array were sorted.
+    /// - Any value at position `i < index` will be less than or equal to any value at a position
+    ///   `j > index`.
+    ///
+    /// As an example, `v.sort_at_index(v.len() / 2)` can be used to find the median of the array.
```

Comment:
This seems kinda out-of-place here.

Reply:
If you mention the median in the examples then this statement is somewhat redundant, though not everyone always looks at examples. I would place it along with the "kth element" statement after this paragraph.

Reply:
My sense is that adding the note about the median isn't terribly relevant here, provided it's (A) in the example and (B) won't necessarily return the median (e.g. for slices with an even number of elements).